assets/app.js

```diff
@@ -1210,4 +1210,116 @@ document.addEventListener('DOMContentLoaded', () => {
   if (typeof FooterMenuToggle !== 'undefined') {
     new FooterMenuToggle();
   }
-});
\ No newline at end of file
+});
+
+class GoogleSheetStocklist extends HTMLElement {
+  constructor() {
+    super();
+    this.apiKey = 'AIzaSyAlMLzJhbzGL9b_w3VOa43C3MIG3u3rYjw';
+    this.sheetId = '1C0RcxqT6jsdq5nXwLuVcszNJ5M8ONywCTStvGvI7klk';
+    this.range = 'Sheet1';
+    this.url = `https://sheets.googleapis.com/v4/spreadsheets/${this.sheetId}/values/${this.range}?key=${this.apiKey}`;
+    this.allRows = [];
+    this.chunkSize = 10;
+    this.currentIndex = 0;
+    this.observer = new IntersectionObserver(entries => this.onIntersect(entries), {
+      root: null,
+      rootMargin: "0px",
+      threshold: 1
+    });
+  }
+  connectedCallback() {
+    this.innerHTML = `
+      <div id="sheetData" class="store-list-container"></div>
+      <div id="loading" class="loading">Loading...</div>
+    `;
+    this.fetchAndInitializeData();
+  }
+  disconnectedCallback() {
+    this.observer.disconnect();
+  }
+  async fetchAndInitializeData() {
+    try {
+      this.showLoading(true);
+      const response = await fetch(this.url);
+      const data = await response.json();
+      this.showLoading(false);
+      if (!data.values || data.values.length < 2) {
+        console.error('Error: No data found or only header row available.');
+        return;
+      }
+      this.allRows = data.values.slice(1);
+      this.loadNextChunk();
+    } catch (error) {
+      this.showLoading(false);
+      console.error('Error fetching Google Sheets data:', error);
+    }
+  }
+  loadNextChunk() {
```

Comment:
lazy loading is not required as the data is already fetched.
render all cards once the data is fetched